### DISCUSSION TITLE
More info about getSchemaNotFromBuildSchemaFunction in examples

### DISCUSSION THREAD

**Discussion by pedroabreu** [MichalLytek/type-graphql#1105] [Q&A]
In the example to generate the schema, it refers a function `getSchemaNotFromBuildSchemaFunction`

https://github.com/MichalLytek/type-graphql/blob/776e458c584e89660e3d8449793b7a6b2c667e0b/docs/emit-schema.md

What would that look like? I can only see from the code the actual `buildSchema`. Is there more detail that could be added here? For context I'm looking into generating the schema so I can use gql-codegen to generate the types for client consumption.

**Comment by MichalLytek** [comment]
You can use it with any other graphql tool, like grahql-js, nexus, schema stiching, etc.

**Reply by pedroabreu** [reply]
But none of those tools would generate a schema from .ts files - as far as I know. So I would still need to use the `buildSchema`, is that correct?

**Reply by MichalLytek** [reply]
It doesn't matter - `getSchemaNotFromBuildSchemaFunction` means you can even do `const schema = new GraphQLSchema(...)` - `emitSchemaDefinitionFile` works with everything, not only TypeGraphQL.

**Reply by MichalLytek** [reply]
But yes, for TypeGraphQL, you always use `buildSchema`, but you can `e.g.` import that from other package, so this function let you do `emitSchemaDefinitionFile(schema)` when you don't have control over `buildSchema` and you can't set `emitSchemaFile: true,`.
